odo/backends/dask_array.py

```diff
@@ -0,0 +1,74 @@
+from __future__ import absolute_import, division, print_function
+
+import numpy as np
+from datashape import discover
+from toolz import merge, accumulate
+from datashape.dispatch import dispatch
+from datashape import DataShape
+from operator import add
+import itertools
+
+from dask.array.core import rec_concatenate, Array, getem, get, names, from_array
+from dask.core import flatten
+from dask.compatibility import long
+
+from ..append import append
+from ..convert import convert
+from ..resource import resource
+
+@discover.register(Array)
+def discover_dask_array(a, **kwargs):
+    block = a._get_block(*([0] * a.ndim))
+    return DataShape(*(a.shape + (discover(block).measure,)))
+
+
+arrays = [np.ndarray]
+try:
+    import h5py
+    arrays.append(h5py.Dataset)
+
+    @dispatch(h5py.Dataset, (int, long))
+    def resize(x, size):
+        s = list(x.shape)
+        s[0] = size
+        return resize(x, tuple(s))
+
+    @dispatch(h5py.Dataset, tuple)
+    def resize(x, shape):
+        return x.resize(shape)
+except ImportError:
+    pass
+try:
+    import bcolz
+    arrays.append(bcolz.carray)
+
+    @dispatch(bcolz.carray, (int, long))
+    def resize(x, size):
+        return x.resize(size)
+except ImportError:
+    pass
+
+
+
+@convert.register(Array, tuple(arrays), cost=0.01)
+def array_to_dask(x, name=None, blockshape=None, **kwargs):
+    return from_array(x, blockshape=blockshape, name=name, **kwargs)
```

Comment:
That's valid.  from_array might not raise NotImplementedError though.  More
likely ValueError.  The network mechanism in odo only skips on
NotImplementedError and raises on all others.

On Wed, Mar 11, 2015 at 10:13 AM, Blake Griffith <notifications@github.com>
wrote:

> In odo/backends/dask_array.py
> <https://github.com/ContinuumIO/odo/pull/128#discussion_r26232979>:
>
> > +    pass
> > +try:
> > +    import bcolz
> > +    arrays.append(bcolz.carray)
> > +
> > +    @dispatch(bcolz.carray, (int, long))
> > +    def resize(x, size):
> > +        return x.resize(size)
> > +except ImportError:
> > +    pass
> > +
> > +
> > +
> > +@convert.register(Array, tuple(arrays), cost=0.01)
> > +def array_to_dask(x, name=None, blockshape=None, **kwargs):
> > +    return from_array(x, blockshape=blockshape, name=name, **kwargs)
>
> That should be raised in dask by the dask.core.from_array function, not
> here. Right?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ContinuumIO/odo/pull/128/files#r26232979>.
>